docs/en_US/NAS/retiarii/Tutorial.rst

```diff
@@ -0,0 +1,171 @@
+
+*We are working on migrating* `our previous NAS framework <../Overview.rst>`__ *to Retiarii framework. Thus, this feature is still experimental. We recommend users to try the new framework and provide your valuable feedback for us to improve it. You can also use the old framework if you like.*
+
+.. contents::
+
+There are mainly two steps to start an experiment for your nueral architecture search task. First, define the model space you want to explore. Second, choose a search method to explore your defined model space.
+
+Define your Model Space
+-----------------------
+
+Model space is defined by users to express a set of models that users want to explore, and believe good-performing models are included in those models. In our framework, a model space is defined with two parts: a base model and possible mutations on the base model.
+
+Define Base Model
+^^^^^^^^^^^^^^^^^
+
+Defining a base model is almost the same as defining a PyTorch (or TensorFlow) model. There are only two small differences.
+
+* Replace PyTorch ``nn`` with our wrapped ``nn`` for PyTorch modules. For example, replace ``import torch.nn as nn`` with ``import nni.retiarii.nn.pytorch as nn``
+* Add the decorator ``@register_module()`` above user defined PyTorch module.
+
+Below is a simple example code of how to define a base model.
+
+.. code-block:: python
+
+  import torch.nn.functional as F
+  import nni.retiarii.nn.pytorch as nn
+  from nni.retiarii import register_module
+
+  @register_module()
+  class MyModule(nn.Module):
+    def __init__(self):
+      super().__init__()
+      self.conv = nn.Conv2d(32, 1, 5)
+      self.pool = nn.MaxPool2d(kernel_size=2)
+    def forward(self, x):
+      return self.pool(self.conv(x))
+
+  @register_module()
+  class Model(nn.Module):
+    def __init__(self):
+      super().__init__()
+      self.mymodule = MyModule()
+    def forward(self, x):
+      return F.relu(self.mymodule(x))
+
+Users can refer to `Darts base model <https://github.com/microsoft/nni/blob/master/test/retiarii_test/darts/darts_model.py>`__ and `Mnasnet base model <https://github.com/microsoft/nni/blob/master/test/retiarii_test/mnasnet/base_mnasnet.py>`__ for more complete examples.
+
+Define Model Mutations
+^^^^^^^^^^^^^^^^^^^^^^
+
+A base model is only one concrete model not a model space. To define model space, we provide APIs and primitives for users to express how the base model can be mutated.
+
+**Express mutations in an inlined manner**
+
+For easy usability and also backward compatibility, we provide some APIs for users to easily express possible mutations during defining a base model. The APIs can be used just like PyTorch module.
+
+* ``nn.LayerChoice``. It allows users to put several candidate operations (e.g., PyTorch modules), one is chosen in each explored model.
```

Comment:
Which ``nn``  should be used? ``torch.nn`` or ``retiarii.nn.pytorch``?

Reply:
it should be `retiarii.nn.pytorch`, i will make it clear, thanks!